receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -0,0 +1,316 @@
+	var alert model.Alert
+
+	err := json.Unmarshal(payload, &alert)
+	if err != nil {
+		return nil, err
+	}
+
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	logRecord := resourceLogs.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+	logRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(now))
+	logRecord.SetTimestamp(timestampFromAlert(alert))
+	logRecord.SetSeverityNumber(severityFromAlert(alert))
+	logRecord.Body().SetStringVal(string(payload))
+
+	resourceAttrs := resourceLogs.Resource().Attributes()
+	resourceAttrs.InsertString("mongodbatlas.group.id", alert.GroupID)
+	resourceAttrs.InsertString("mongodbatlas.alert.config.id", alert.AlertConfigID)
+	insertStringToMapNotNil(resourceAttrs, "mongodbatlas.cluster.name", alert.ClusterName)
+	insertStringToMapNotNil(resourceAttrs, "mongodbatlas.replica_set.name", alert.ReplicaSetName)
+
+	attrs := logRecord.Attributes()
+	// These attributes are always present
+	attrs.InsertString("event.name", alert.EventType)
+	attrs.InsertString("message", alert.HumanReadable)
+	attrs.InsertString("status", alert.Status)
+	attrs.InsertString("created", alert.Created)
+	attrs.InsertString("updated", alert.Updated)
+	attrs.InsertString("id", alert.ID)
+
+	// These attributes are optional and may not be present, depending on the alert type.
+	insertStringToMapNotNil(attrs, "metric.name", alert.MetricName)
+	insertStringToMapNotNil(attrs, "type_name", alert.TypeName)
+	insertStringToMapNotNil(attrs, "user_alias", alert.UserAlias)
+	insertStringToMapNotNil(attrs, "last_notified", alert.LastNotified)
+	insertStringToMapNotNil(attrs, "resolved", alert.Resolved)
+	insertStringToMapNotNil(attrs, "acknowledgement.comment", alert.AcknowledgementComment)
+	insertStringToMapNotNil(attrs, "acknowledgement.username", alert.AcknowledgementUsername)
+	insertStringToMapNotNil(attrs, "acknowledgement.until", alert.AcknowledgedUntil)
+
+	if alert.CurrentValue != nil {
+		attrs.InsertDouble("metric.value", alert.CurrentValue.Number)
+		attrs.InsertString("metric.units", alert.CurrentValue.Units)
+	}
+
+	if alert.HostNameAndPort != nil {
+		host, portStr, err := net.SplitHostPort(*alert.HostNameAndPort)
+		if err != nil {
+			return nil, fmt.Errorf("failed to split host:port %s: %w", *alert.HostNameAndPort, err)
+		}
+
+		port, err := strconv.ParseInt(portStr, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse port %s: %w", portStr, err)
+		}
+
+		attrs.InsertString("net.peer.name", host)
+		attrs.InsertInt("net.peer.port", port)
+
+	}
+
+	return &logs, nil
+}
+
+func timestampFromAlert(a model.Alert) pcommon.Timestamp {
+	if time, err := time.Parse(time.RFC3339, a.Updated); err == nil {
+		return pcommon.NewTimestampFromTime(time)
+	}
+
+	if time, err := time.Parse(time.RFC3339, a.Created); err == nil {
+		return pcommon.NewTimestampFromTime(time)
+	}
+
+	return pcommon.Timestamp(0)
+}
+
+func severityFromAlert(a model.Alert) plog.SeverityNumber {
```

Comment:
Is there a doc link we can post as a comment in here just for a reference for any one who looks at this?